packages/blocks-ui/src/panels/blocks-listing.js

```diff
@@ -9,6 +9,70 @@ import { useEditor } from '../providers/editor'
 import { useBlocks } from '../providers/blocks'
 import { useScope } from '../providers/scope'
 
+// used as a wrapper around like block types
+const BlockSorting = ({ title, children }) => {
+  const [toggle, setToggle] = useState(true)
+  const [height, setHeight] = useState()
```

Comment:
You can use the custom hook in called `useElementSize` which gets the height/width of a ref for you and handles all the event listeners. We are also using this hook in `canvas.js` if you want to take a look at an example. That way, you can get rid of this `useEffect`.